packages/terra-search-field/src/SearchField.jsx

```diff
@@ -183,36 +225,57 @@ class SearchField extends React.Component {
       inputAttributes,
       ...customProps
     } = this.props;
+
+    const cancelButtonVisible = !defaultValue && !this.state.searchText ? 'input-clear' : undefined;
+
+    const cancelButtonClassNames = cx([
+      'clear',
+    ]);
+
+    const inputClassNames = cx([
+      'input',
+      cancelButtonVisible,
+    ]);
+
     const searchFieldClassNames = cx([
       'searchfield',
       { block: isBlock },
       customProps.className,
     ]);
 
-    const valueTemp = {};
-    if (value !== undefined) {
-      valueTemp.value = value;
-    } else {
-      valueTemp.defaultValue = defaultValue;
-    }
-
     const inputText = this.context.intl.formatMessage({ id: 'Terra.searchField.search' });
     const buttonText = this.context.intl.formatMessage({ id: 'Terra.searchField.submit-search' });
-    const additionalInputAttributes = Object.assign({ 'aria-label': inputText }, inputAttributes, valueTemp);
+    const additionalInputAttributes = Object.assign({ 'aria-label': inputText }, inputAttributes);
 
     return (
       <div {...customProps} className={searchFieldClassNames}>
         <Input
-          className={cx('input')}
+          defaultValue={defaultValue}
+          className={inputClassNames}
           type="search"
           placeholder={placeholder}
           onChange={this.handleTextChange}
+          onFocus={this.handleInputFocus}
           disabled={isDisabled}
           aria-disabled={isDisabled}
           onKeyDown={this.handleKeyDown}
           refCallback={inputRefCallback}
+          value={defaultValue ? undefined : this.state.searchText}
           {...additionalInputAttributes}
         />
+        { this.previouslyFocusedInput && this.state.searchText
+          ? (
+            <Button
+              className={cancelButtonClassNames}
+              onClick={this.handleClear}
+              text="cancel"
+              icon={Clear}
```

Comment:
We'll want this icon to be a background image we can theme. Take a look at [how the icon is set on this button](https://github.com/cerner/terra-core/blob/master/packages/terra-dialog/src/Dialog.jsx#L54) and [the associated styles for it](https://github.com/cerner/terra-core/blob/master/packages/terra-dialog/src/Dialog.module.scss#L31-L38) as an example.

Reply:
Finally working: [3c644fc](https://github.com/cerner/terra-core/pull/2317/commits/3c644fc526ca5b970747607ce166f4e43ecc93cf)